[RHIDP-7712]: Add configuration files reference to the Configuring title (#2257)
redhat-developer/red-hat-developers-documentation-rhdh: modules/configure_configuring-rhdh/con-rhdh-secrets.adoc

:_mod-docs-content-type: CONCEPT

[id="rhdh-secrets_{context}"]
= {product} secrets

[role="_abstract"]
{product-short} uses Kubernetes secrets to store sensitive values such as authentication credentials, backend secrets, and database passwords. The `{my-app-config-file}` file references these values through `$\{VAR_NAME}` environment variable substitution.

The following are common secrets used by {product-short}:

`my-rhdh-secrets`:: The main {product-short} secrets containing authentication provider client IDs and secrets, the `BACKEND_SECRET`, and other sensitive credentials.

`my-rhdh-database-certificates-secrets`:: Optional. Contains external PostgreSQL TLS certificates such as `postgres-crt.pem`, `postgres-ca.pem`, and `postgres-key.key`.

To create secrets, author the secret values in a local file (for example, `my-rhdh-secrets.txt`), then create a Kubernetes secret by running `oc create secret generic`.

You can provision secrets using either of the following methods:

> * With the Operator: reference secrets in the `{product-custom-resource-type}` custom resource by using `spec.application.extraEnvs.secrets` to inject secrets as environment variables, or `spec.application.extraFiles.secrets` to mount secrets as files (for example, TLS certificates).

Reviewer: ```suggestion
* With the Operator: Reference secrets in the `{product-custom-resource-type}` custom resource by using the `spec.application.extraEnvs.secrets` field to inject secrets as environment variables, or the `spec.application.extraFiles.secrets` field to mount secrets as files, such as, TLS certificates.
```

> * With the Helm chart: reference secrets by using `upstream.backstage.extraEnvs.secrets`.

Reviewer 2: ```suggestion
* With the Helm chart: reference secrets by using `upstream.backstage.extraEnvVarsSecrets` or `upstream.backstage.extraEnvVars` (using a [`secretKeyRef`](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables)). Secrets can also be mounted as files using both `upstream.backstage.extraVolumes` and `upstream.backstage.extraVolumeMounts`.
```

Just to clarify, but feel free to rephrase.

> * With the Helm chart: reference secrets by using `upstream.backstage.extraEnvVarsSecrets` or `upstream.backstage.extraEnvVars` (using a [`secretKeyRef`](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables)). Secrets can also be mounted as files using both `upstream.backstage.extraVolumes` and `upstream.backstage.extraVolumeMounts`.

Reviewer: ```suggestion
* With the Helm chart: Reference secrets by using the `upstream.backstage.extraEnvVarsSecrets` or `upstream.backstage.extraEnvVars` field with a [`secretKeyRef`](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables)). You can also mount secrets as files by using both the `upstream.backstage.extraVolumes` and `upstream.backstage.extraVolumeMounts` values.
```
